File: ambeth-integrity/de.osthus.classbrowser.csharp/CsharpClassbrowser/CsharpClassbrowser/AnnotationParamInfo.cs
using System;

namespace CsharpClassbrowser
{
    /// <summary>
    /// Holds the information for annotation params.
    /// </summary>
    public class AnnotationParamInfo : INamed
    {

        // ============================================================================================
        #region Constructors
        // ============================================================================================

        /// <summary>
        /// Create a new instance.
        /// </summary>
        /// <param name="returnType">Annotation type; mandatory</param>
        /// <param name="parameterTypes">Parameters; mandatory</param>
        public AnnotationParamInfo(string name, string type, object defaultValue, object currentValue)
        {
            if (string.IsNullOrWhiteSpace(name) || string.IsNullOrWhiteSpace(type))
            {
                throw new ArgumentNullException("Mandatory annotation param info value missing!");
            }

            this.Name = name;
            this.Type = type;
            this.DefaultValue = defaultValue;
            this.CurrentValue = currentValue;
        }

        #endregion

        // ============================================================================================
        #region Properties
        // ============================================================================================

        public string Name { get; internal set; }
        public string Type { get; internal set; }
        public object DefaultValue { get; internal set; }
        public object CurrentValue { get; internal set; }

        #endregion

    }
}

File: ambeth-integrity/com.koch.classbrowser.java/src/main/java/com/koch/classbrowser/java/AnnotationParamInfo.java
package com.koch.classbrowser.java;

import org.apache.commons.lang3.StringUtils;
import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

/**
 * Holds the information for annotation params.
 * 
 * @author jochen.hormes
 */
public class AnnotationParamInfo implements INamed
{
	// ---- VARIABLES ----------------------------------------------------------

	private String name;

	private String type;

	private Object defaultValue;

	private Object currentValue;

	// ---- CONSTRUCTORS -------------------------------------------------------

	/**
	 * Create a new instance.
	 * 
	 * @param name
	 *            Annotation parameter name; mandatory
	 * @param type
	 *            Annotation parameter type; mandatory
	 * @param defaultValue
	 *            Annotation parameter default value; may be null
	 * @param currentValue
	 *            Annotation parameter current value; may be null
	 */
	public AnnotationParamInfo(String name, String type, Object defaultValue, Object currentValue)
	{
		if (StringUtils.isBlank(name) || StringUtils.isBlank(type))
		{
			throw new IllegalArgumentException("Mandatory annotation param info value missing!");
		}

		this.name = name;
		this.type = type;
		this.defaultValue = defaultValue;
		this.currentValue = currentValue;
	}

	// ---- GETTER/SETTER METHODS ----------------------------------------------

	/**
	 * {@inheritDoc}
	 */
	@Override
	public String getName()
	{
		return name;
	}

	public String getType()
	{
		return type;
	}

	public Object getDefaultValue()
	{
		return defaultValue;
	}

	public Object getCurrentValue()
	{
		return currentValue;
	}

	// ---- METHODS ------------------------------------------------------------

	@Override
	public String toString()
	{
		return new ToStringBuilder(this, ToStringStyle.SHORT_PREFIX_STYLE).append("param name", type).append("param type", type)
				.append("param default value", defaultValue).append("param current value", currentValue).toString();
	}
}
